kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -428,8 +504,10 @@ private Optional<LogSegment> getLogSegmentFrom(
         Engine engine,
```

Comment:
Looks like there is only one callsite of this function. My preference would be to inline it or change the name to indicate that the function will not return CC commits. My preference would be just to inline it to reduce the complexity. What do you think @vkorukanti?